surefire-api/src/main/java/org/apache/maven/surefire/api/booter/MasterProcessChannelEncoder.java

```diff
@@ -37,7 +38,7 @@
 
     void onJvmExit();
 
-    void systemProperties( Map<String, String> sysProps );
+    void systemProperties( Map<String, String> sysProps, RunMode runMode, Long testRunId );
```

Comment:
method name is not descriptive .... what does do .. read/write/verify properties? 
Now new params - what for? 

documentation is needed

Reply:
@Tibor17 can you add some documentation for this method ... or you have such plan?

Reply:
I agree and so I have added the Javadoc, pls see the commit [2840bfa](https://github.com/apache/maven-surefire/pull/475/commits/2840bfaf14abd4574620ce6825047230636d928c).

Reply:
The testRunId says - this is the test set.
The runMode says - normal mode, or re-run.